core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -888,7 +897,16 @@ public void workerRegister(long workerId, List<String> storageTiers,
       blocks.addAll(blockIds);
     }
 
+    Instant waitWorkerBlock = Instant.now();
+    // This synchronized section is guarding two workers registering with the same NetAddress
+    // This should happen with a very low possibility
+    // And should not be the performance bottleneck
     synchronized (worker) {
```

Comment:
This synchronized section should have very little contention.

Reply:
Unless we trigger contention over the same worker object. The possible situations include:
1. register the same worker in a high volume.
This test makes no sense in reality. However it's probably the easiest way if we want to poke the bottleneck on this critical section. 

2. In many other calls they also do `synchronized(worker)`, for example `commitBlock`
It can be harder to simulate those calls. If what matters is only this synchronized lock, then we do Point 1.